circlecheck.lic

```diff
@@ -604,17 +605,14 @@ def display_requirements(requirements, level_target, brief, pretty_print)
 
       # Calculate the percent remaining to next skill rank
       # so we can display fractional amounts. Format so 
-      # we can easily append to current rank.
-      percent_remaining = 100 - DRSkill.getpercent(sub_skill).to_i
-      if percent_remaining.to_s.size < 2
-        percent_remaining = "0" + percent_remaining.to_s
-      elsif percent_remaining == 100
-        percent_remaining = 00
-      end
+      # we can easily append decimal to current rank.
+      subtraction = (100 - DRSkill.getpercent(sub_skill).to_f)/100.00
+      percent_remaining = format('%.2f', subtraction).sub(/^./, '')
 
       # Formulate the needed ranks to target circle including 
       # fractional amount. Also prep total target ranks for proper display.
-      needed_ranks = (missing_ranks -1).to_s + "." + percent_remaining.to_s
+      subtraction == 1.0 ? needed_ranks = missing_ranks : needed_ranks = missing_ranks - 1
```

Comment:
💡 It took me a minute to understand this syntax. Thoughts on refactoring to one of these alternatives such that there's exactly one assignment per statement and the "right hand side" is only deciding on a return value?
```ruby
      needed_ranks = (subtraction == 1.0) ? missing_ranks : missing_ranks - 1
```
```ruby
      needed_ranks = missing_ranks
      needed_ranks -= 1 if subtraction == 1.0
```

Reply:
I was exploring the use of Ruby's single-line if/else syntax. It's hard for me to read.

Ok to just use a standard if/else?

```
if subtraction == 1.0
    needed_ranks = missing_ranks
else
    needed_ranks = missing_ranks - 1
end
```

Reply:
Yep, readability > syntax magic